Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -56,10 +58,42 @@ def check_python_framework_links(lib)
     EOS
   end
 
+  def check_linkage
+    return unless formula.prefix.directory?
+    keg = Keg.new(formula.prefix)
+    checker = LinkageChecker.new(keg, formula)
+    s = ""
+
+    if checker.broken_dylibs?
+      s += <<-EOS.undent
+        Broken dylib links found:
+          #{checker.broken_dylibs.to_a * "\n          "}
+
+        It is likely the installation was broken.
+        Please check formula and make sure it is built from source.
+      EOS
+    end
+
+    if checker.undeclared_deps?
+      s += <<-EOS.undent
+        Undeclared dependencies found:
+          #{checker.undeclared_deps * "\n          "}
```

Comment:
May I suggest to list in here rather than `depends_on` below(we could offer one example for the first formula). Because authors are required to check whether they are intended.